modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/introspection/RuntimeSnapshot.java

```diff
@@ -403,29 +402,27 @@ protected void initOperations() {
         }
     }
 
+    protected void addOperation(OperationInfo opi) {
+        this.operations.put(opi.getName(), opi);
+        List<String> aliases = opi.getAliases();
+        if (aliases != null) {
+            aliases.forEach(a -> this.operations.put(a, opi));
+        }
+    }
+
     @Override
     public OperationInfo getOperation(String id) {
+        String finalId = id;
         if (id.startsWith(OperationInfo.ARTIFACT_PREFIX)) {
-            id = id.substring(OperationInfo.ARTIFACT_PREFIX.length());
-        }
-        for (OperationInfo op : getOperations()) {
-            if (op.getName().equals(id)) {
-                return op;
-            }
-
-            String finalId = id;
-            Optional<String> first = op.getAliases().stream().filter(s -> s.equals(finalId)).findFirst();
-            if (first.isPresent()) {
-                return op;
-            }
+            finalId = id.substring(OperationInfo.ARTIFACT_PREFIX.length());
         }
-        return null;
+        return operations.get(finalId);
     }
 
     @Override
     public List<OperationInfo> getOperations() {
         initOperations();
-        return Collections.unmodifiableList(operations);
+        return operations.values().stream().collect(Collectors.toList());
```

Comment:
Don't we want to keep the unmodifiable list? Otherwise `new ArrayList(operations.values())` is probably better as we do nothing on the stream.

Reply:
We only want to prevent changes to the underlying inner map/list, i'll make the change :+1: